### ISSUE TITLE
Java 20 support

### ISSUE THREAD

**Issue by Hongbo-Miao** [detekt/detekt#6031]
`1.23.0-RC1` works for Java 19. It would be great to support Java 20, thanks! 😃

Currently it shows

```sh
Execution failed for task ':app:detekt'.
> Invalid value (20) passed to --jvm-target, must be one of [1.6, 1.8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
```

**Comment by cortinico** [comment]
From what I can see, we're constrained by the Kotlin version (1.8.20 doesn't allow to `jvmTarget` to be `20`):
https://github.com/JetBrains/kotlin/blob/01a381b892f940d7995ec98e05a8a62011064b2b/compiler/config.jvm/src/org/jetbrains/kotlin/config/JvmTarget.kt#L22-L39

I see they added support for `20` on main so this will come with the upcoming Kotlin version.

**Comment by 3flex** [comment]
Fixed by #6258.

I've opened #6277 to update the CLI help output to include Java 20 target support.
